docs/rector_rules_overview.md

```diff
@@ -14170,6 +14171,108 @@ Adds a new `$services->set(...)` call to PHP Config
 
 ## RemovingStatic
 
+### DesiredClassTypeToDynamicRector
+
+Change full static service, to dynamic one
+
+- class: `Rector\RemovingStatic\Rector\Class_\DesiredClassTypeToDynamicRector`
+
+```diff
+ class AnotherClass
+ {
++    /**
++     * @var SomeClass
++     */
++    private $someClass;
++
++    public fuction __construct(SomeClass $someClass)
++    {
++        $this->someClass = $someClass;
++    }
++
+     public function run()
+     {
+         SomeClass::someStatic();
+     }
+ }
+
+ class SomeClass
+ {
+-    public static function run()
++    public function run()
+     {
+-        self::someStatic();
++        $this->someStatic();
+     }
+
+-    private static function someStatic()
++    private function someStatic()
+     {
+     }
+ }
+```
+
+<br>
+
+### DesiredPropertyClassMethodTypeToDynamicRector
+
+Change defined static properties and methods to dynamic
+
+- class: `Rector\RemovingStatic\Rector\Property\DesiredPropertyClassMethodTypeToDynamicRector`
+
+```diff
+ final class SomeClass
+ {
+-    public static $name;
++    public $name;
+
+-    public static function go()
++    public function go()
+     {
+     }
+ }
+```
+
+<br>
+
+### DesiredStaticCallTypeToDynamicRector
+
+Change defined static service to dynamic one
+
+- class: `Rector\RemovingStatic\Rector\StaticCall\DesiredStaticCallTypeToDynamicRector`
+
+```diff
+ final class SomeClass
+ {
+     public function run()
+     {
+-        SomeStaticMethod::someStatic();
++        $this->someStaticMethod::someStatic();
+     }
+ }
+```
+
+<br>
+
+### DesiredStaticPropertyFetchTypeToDynamicRector
+
+Change defined static service to dynamic one
+
+- class: `Rector\RemovingStatic\Rector\StaticPropertyFetch\DesiredStaticPropertyFetchTypeToDynamicRector`
+
+```diff
+ final class SomeClass
+ {
+     public function run()
+     {
+-        SomeStaticMethod::$someStatic;
++        $this->someStaticMethod::$someStatic;
+     }
+ }
+```
+
+<br>
+
```

Comment:
@Kerrialn Today, I've just split the single services for changing static to 4. One for each node. Now it should be easier to re-use them :)

The configuratoin has moved from rule to the `TYPES_TO_REMOVE_STATIC_FROM` parameter, e.g. like this:

```php
<?php

declare(strict_types=1);

use Rector\Core\Configuration\Option;
use Rector\RemovingStatic\Rector\Property\DesiredPropertyClassMethodTypeToDynamicRector;
use Rector\RemovingStatic\Tests\Rector\Property\DesiredPropertyClassMethodTypeToDynamicRector\Fixture\StaticProperty;
use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;

return static function (ContainerConfigurator $containerConfigurator): void {
    $parameters = $containerConfigurator->parameters();
    $parameters->set(Option::TYPES_TO_REMOVE_STATIC_FROM, [StaticProperty::class]);

    $services = $containerConfigurator->services();
    $services->set(DesiredPropertyClassMethodTypeToDynamicRector::class);
};
```